File: PatientApp/patientapp/lib/models/patient.dart
import 'package:flutter_guid/flutter_guid.dart';
import 'package:patientapp/models/phone.dart';
import 'package:flutter/material.dart';

class Patient {
  Guid patientId;
  String firstName;
  String lastName;
  String email;
  bool isDeleted;
  List<Phone> phones;

  Patient({
    Guid patientId,
    String firstName = '',
    @required String lastName,
    String email = '',
    bool isDeleted = false,
    List<Phone> phones,
  })  : this.patientId = patientId,
        this.firstName = firstName,
        this.lastName = lastName,
        this.email = email,
        this.isDeleted = isDeleted,
        this.phones = phones;

  factory Patient.fromJson(Map<String, dynamic> json) {
    List<Phone> phones =
        (json['phones'] as List).map((phone) => Phone.fromJson(phone)).toList();
    Patient patient = new Patient(
      patientId: new Guid(json['patientId']),
      firstName: json['firstName'],
      lastName: json['lastName'],
      email: json['email'],
      isDeleted: json['isDeleted'],
    );

    patient.phones = phones;
    return patient;
  }

  Map<String, dynamic> toJson() {
    return {
      if (patientId != null) 'patientId': patientId.toString(),
      'firstName': firstName,
      'lastName': lastName,
      'email': email,
      'isDeleted': isDeleted,
    };
  }
}

File: PatientApp/PatientAppApi/PatientApp/Models/Patient.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.EntityFrameworkCore.Metadata.Internal;
using System.ComponentModel.DataAnnotations;

namespace PatientApp.Models
{
    public class Patient
    {
        [Key]
        public Guid PatientId { get; set; }
        public string FirstName { get; set; }
        [Required]
        public string LastName { get; set; }
        public string Email { get; set; }
        public List<Phone> Phones { get; set; }
        public bool IsDeleted { get; set; } = false;
        
        
    }
}
